ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -1676,12 +1691,35 @@ - (BOOL)isEqual:(id)object {
              self.autocapitalizationType == castObject.autocapitalizationType &&
              self.autocorrectionType == castObject.autocorrectionType &&
              self.spellCheckingType == castObject.spellCheckingType &&
+             self.keyboardType == castObject.keyboardType &&
              self.multipleLines == castObject.multipleLines));
 }
 
 @end
 
 
+#pragma mark - ORKEmailAnswerFormat
+
+@implementation ORKEmailAnswerFormat
+
+- (Class)questionResultClass {
+    return [ORKTextQuestionResult class];
+}
+
+- (ORKAnswerFormat *) impliedAnswerFormat{
+    ORKTextAnswerFormat *answerFormat = [ORKTextAnswerFormat textAnswerFormatWithMaximumLength:0];
+    answerFormat.keyboardType = UIKeyboardTypeEmailAddress;
+    answerFormat.multipleLines = NO;
+    answerFormat.spellCheckingType = UITextSpellCheckingTypeNo;
+    answerFormat.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    answerFormat.autocorrectionType = UITextAutocorrectionTypeNo;
+    answerFormat.isEmail = YES;
+    return answerFormat;
```

Comment:
Worth to cache this `answerFormat` object, since it cost many lines to construct.